pulp_ansible/app/tasks/collections.py

```diff
@@ -131,10 +131,42 @@ def import_collection(artifact_pk):
             collection, created = Collection.objects.get_or_create(
                 namespace=collection_info["namespace"], name=collection_info["name"]
             )
-            collection_version = CollectionVersion(
-                collection=collection, version=collection_info["version"]
-            )
+            if created:
+                CreatedResource.objects.create(content_object=collection)
+
+            dependencies = collection_info.pop("dependencies")
+            tags = collection_info.pop("tags")
+            tags = ["onetag", "anothertag"]
+
+            # Mazer returns many None values. We need to let the default='' in models.py prevail
+            for key in [
+                "description",
+                "documentation",
+                "homepage",
+                "issues",
+                "license_file",
+                "readme",
+                "repository",
+            ]:
+                if collection_info[key] is None:
+                    collection_info.pop(key)
+
+            collection_version = CollectionVersion(collection=collection, **collection_info)
             collection_version.save()
+
+            for dependency in dependencies.values():
+                dependency_split = dependency.split(".")
+                namespace = dependency_split[0]
+                name = dependency_split[1]
+                found_dependency = CollectionVersion.objects.get(namespace=namespace, name=name)
```

Comment:
How is that supposed to work? There can be >1 version matching dependency specifier.

Reply:
It's not going to work. I didn't realize the versionspec part of the design until later. This will be changing very soon (before merging).